File: Java(eclipse)/workspace/書籍検索アプリ/src/main/java/todo/model/CompositeLoadData.java
package todo.model;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import todo.model.beans.BookDataBean;

public class CompositeLoadData implements ILoadData {

    private final List<ILoadData> loaders;

    public CompositeLoadData(ILoadData... loaders) {
    	
        this.loaders = Arrays.asList(loaders);
    }

    @Override
    public BookDataBean[] BookDataBeanGet() {
    	
        List<BookDataBean> result = new ArrayList<>();

        for (ILoadData loader : loaders) {
        	
            BookDataBean[] data = loader.BookDataBeanGet();
            if (data != null) {
            	
                result.addAll(Arrays.asList(data));
            }
        }

        return result.toArray(BookDataBean[]::new);
    }
}

File: 1_C#(VisualStudio)/1_workspace_ASP.NETCoreMVC/1_書籍検索アプリ_移植元Java/Models/CompositeLoadData.cs
using 書籍検索アプリ_移植元Java.Models.beans;

namespace 書籍検索アプリ_移植元Java.Models
{
    public class CompositeLoadData : ILoadData {

        private readonly List<BookDataBean> resultBeans;

        public CompositeLoadData(List<ILoadData> loaderList) {
    	
    	    resultBeans = new List<BookDataBean>();
            foreach (ILoadData loader in loaderList) {
        	
                BookDataBean[] data = loader.BookDataBeanGet();
                if (data != null) {
            	
            	    resultBeans.AddRange(data); // ← Java の addAll に相当
                }
            }
        }

        public BookDataBean[] BookDataBeanGet() {
    	
            return resultBeans.ToArray();
        }
    }
}